src/averageAges.js

```diff
@@ -55,7 +67,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people
+    .filter(child => people
+      .some(woman => woman.name === child.mother));
+
+  const son = people
+    .filter(child => people
+      .some(woman => child.mother === woman.name && child.sex === 'm'));
+
+  const currentSearch = onlyWithSon ? son : children;
+  const diff = currentSearch
+    .map(child =>
+      child.born - (people.find(woman => woman.name === child.mother)).born);
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

better rename to `ageDifference`, variables should be meaningful.

And better create variable `motherBorn` for example, this way code would be easier to read.